app/services/file-embeddings.ts

```diff
@@ -0,0 +1,384 @@
+        match_count: 10,
+      });
+      
+      if (similar && similar.length > 0) {
+        similarFiles.set(
+          inputFile.file_path,
+          similar
+            .filter((s: any) => s.file_path !== inputFile.file_path)
+            .map((s: any) => ({
+              path: s.file_path,
+              similarity: s.similarity,
+            }))
+        );
+      }
+    }
+    
+    return similarFiles;
+  } catch (error) {
+    console.error('Error finding similar files:', error);
+    return new Map();
+  }
+}
+
+/**
+ * Prepare file content for embedding generation
+ */
+function prepareContentForEmbedding(filePath: string, content: string): string {
+  const maxLength = 8000; // Maximum characters for embedding
+  
+  // Extract the most important parts of the file
+  let prepared = `File: ${filePath}\n\n`;
+  
+  // Add file extension context
+  const ext = filePath.split('.').pop()?.toLowerCase() || '';
+  prepared += `Type: ${getFileType(ext)}\n\n`;
+  
+  // For code files, try to extract important sections
+  if (isCodeFile(filePath)) {
+    // Extract imports/includes
+    const imports = extractImports(content);
+    if (imports) {
+      prepared += `Imports:\n${imports}\n\n`;
+    }
+    
+    // Extract function/class signatures
+    const signatures = extractSignatures(content);
+    if (signatures) {
+      prepared += `Functions/Classes:\n${signatures}\n\n`;
+    }
+    
+    // Add remaining content (truncated)
+    const remainingSpace = maxLength - prepared.length - 100;
+    if (remainingSpace > 0) {
+      const truncatedContent = content.substring(0, remainingSpace);
+      prepared += `Content:\n${truncatedContent}`;
+    }
+  } else {
+    // For non-code files, just include the content
+    prepared += content.substring(0, maxLength - prepared.length);
+  }
+  
+  return prepared;
+}
+
+/**
+ * Check if a file is a code file
+ */
+function isCodeFile(filePath: string): boolean {
+  const codeExtensions = new Set([
+    'js', 'jsx', 'ts', 'tsx', 'py', 'rb', 'java', 'go', 'rs',
+    'cpp', 'c', 'h', 'hpp', 'cs', 'php', 'swift', 'kt', 'scala',
+    'r', 'vue', 'svelte', 'sql', 'sh', 'bash', 'ps1', 'yaml', 'yml',
+    'json', 'xml', 'html', 'css', 'scss', 'sass', 'less'
+  ]);
+  
+  const ext = filePath.split('.').pop()?.toLowerCase();
+  return ext ? codeExtensions.has(ext) : false;
+}
+
+/**
+ * Get file type description
+ */
+function getFileType(ext: string): string {
+  const typeMap: Record<string, string> = {
+    js: 'JavaScript',
+    jsx: 'React JavaScript',
+    ts: 'TypeScript',
+    tsx: 'React TypeScript',
+    py: 'Python',
+    rb: 'Ruby',
+    java: 'Java',
+    go: 'Go',
+    rs: 'Rust',
+    cpp: 'C++',
+    c: 'C',
+    cs: 'C#',
+    php: 'PHP',
+    swift: 'Swift',
+    kt: 'Kotlin',
+    scala: 'Scala',
+    r: 'R',
+    vue: 'Vue.js',
+    svelte: 'Svelte',
+    sql: 'SQL',
+    sh: 'Shell Script',
+    bash: 'Bash Script',
+    yaml: 'YAML Configuration',
```

Comment:
consider putting this outside the function as it's recreated every time this function is called.